tests/validation/include/detray/validation/straight_line_navigation.hpp

```diff
@@ -0,0 +1,169 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2023 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s)
+#include "detray/intersection/detail/trajectories.hpp"
+#include "detray/propagator/actor_chain.hpp"
+#include "detray/propagator/actors/aborters.hpp"
+#include "detray/propagator/line_stepper.hpp"
+#include "detray/propagator/navigator.hpp"
+#include "detray/propagator/propagator.hpp"
+#include "detray/simulation/event_generator/track_generators.hpp"
+#include "detray/tracks/tracks.hpp"
+#include "detray/utils/inspectors.hpp"
+#include "detray/validation/detail/navigation_check_helper.hpp"
+#include "tests/common/test_base/fixture_base.hpp"
+#include "tests/common/tools/particle_gun.hpp"
+
+// System include(s)
+#include <iostream>
+#include <string>
+
+namespace detray {
+
+/// @brief Test class that runs the straight line navigation check on a given
+///        detector.
+///
+/// @note The lifetime of the detector needs to be guaranteed.
+template <typename detector_t>
+class straight_line_navigation : public test::fixture_base<> {
+
+    using scalar_t = typename detector_t::scalar_type;
+    using transform3_t = typename detector_t::transform3;
+    using ray_t = detail::ray<transform3_t>;
+
+    public:
+    using fixture_type = test::fixture_base<>;
+
+    struct config : public fixture_type::configuration {
+        using trk_gen_config_t =
+            typename uniform_track_generator<ray_t>::configuration;
+
+        std::string m_name{"straight_line_navigation"};
+        trk_gen_config_t m_trk_gen_cfg{};
+
+        /// Getters
+        /// @{
+        const std::string &name() const { return m_name; }
+        trk_gen_config_t &track_generator() { return m_trk_gen_cfg; }
+        const trk_gen_config_t &track_generator() const {
+            return m_trk_gen_cfg;
+        }
+        /// @}
+
+        /// Setters
+        /// @{
+        config &name(const std::string n) {
+            m_name = n;
+            return *this;
+        }
+        /// @}
+    };
+
+    template <typename config_t>
```

Comment:
Why do you need it to be templated?

Reply:
Because I pass in a different config type in sometimes (e.g. the config for a telescope detector to the fixture of the toy detector)